docs/source/gateway/index.rst

```diff
@@ -1,6 +1,1049 @@
+
+To match your user's interaction and security access requirements, adjust the ``openai_api_type`` parameter to represent the preferred security validation model. This will ensure seamless interaction and reliable security for your Azure-OpenAI integration.
+
+- **openai_api_key**: This is the API key for the Azure OpenAI service. It is a required field.
+- **openai_api_type**: When using the Azure integration with OpenAI, this field is required and must be either "azure" or "azuread" depending on the security access protocol being used. This is a required field.
+- **openai_api_base**: This is the base URL for the Azure OpenAI API service that is provided by Azure. It is a required field.
+- **openai_api_version**: The version of the Azure OpenAI service to utilize as specified by a date. It is a required field.
+- **openai_deployment_name**: This is a name of the deployment resource for the Azure OpenAI service. It is a required field.
+- **openai_organization**: This is an optional field to specify the organization in OpenAI.
+
+An example configuration for Azure OpenAI is:
+
+.. code-block:: yaml
+
+    routes:
+      - name: completions
+        route_type: llm/v1/completions
+        model:
+          provider: openai
+          name: gpt-4
+          config:
+            openai_api_type: "azuread"
+            openai_api_key: $AZURE_AAD_TOKEN
+            openai_deployment_name: "{your_deployment_name}"
+            openai_api_base: "https://{your_resource_name}-azureopenai.openai.azure.com/"
+            openai_api_version: "2023-05-15"
+
+
+.. note::
+
+    Azure OpenAI has disctinct features as compared with the direct OpenAI service. For an overview, please see `the comparison documentation <https://learn.microsoft.com/en-gb/azure/cognitive-services/openai/how-to/switching-endpoints>`_.
+
+For specifying an API key, there are three options:
+
+1. (Preferred) Use an environment variable to store the API key and reference it in the YAML configuration file. This is denoted by a "$" symbol before the name of the environment variable.
+2. (Preferred) Define the API key in a file and reference the location of that key-bearing file within the YAML configuration file.
+3. Directly include it in the YAML configuration file.
+
+.. important::
+
+    The use of environment variables or file-based keys is recommended for better security practices. If the API key is directly included in the configuration file, it should be ensured that the file is securely stored and appropriately access controlled.
+    Please ensure that the configuration file is stored in a secure location as it contains sensitive API keys.
+
+AI Gateway Security Considerations
```

Comment:
Can we move this subsection to the bottom and link to it from the `AI Gateway Configuration` section?